prow/interrupts/interrupts.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package interrupts exposes helpers for graceful handling of interrupt signals
+package interrupts
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// RunOrShutdown runs the HTTP server and handles shutting it down
+// gracefully on interrupts. This function is meant to be called in
+// a blocking manner.
+func RunOrShutdown(server *http.Server, gracePeriod time.Duration, logger *logrus.Logger) {
```

Comment:
Metrics and health use this function in a non-blocking manner. Please amend this comment or include comments that explain why metrics and health are exceptions.